Source/GUI/VCL/GUI_Preferences.cpp

```diff
@@ -342,6 +342,20 @@ void __fastcall TPreferencesF::Language_EditClick(TObject *Sender)
 //---------------------------------------------------------------------------
 void __fastcall TPreferencesF::OKClick(TObject *Sender)
 {
+    if (TOSVersion::Major == 10 && TOSVersion::Build >= 22000) {
+        TRegistry* Reg_HideExplorerContextMenu = new TRegistry(KEY_WRITE);
+        try {
+            if (Reg_HideExplorerContextMenu->OpenKey(__T("Software\\MediaArea\\MediaInfo"), true)) {
+                if (CB_InscrireShell->Checked) {
+                    Reg_HideExplorerContextMenu->DeleteValue("HideExplorerContextMenu");
+                }
+                else {
+                    Reg_HideExplorerContextMenu->WriteInteger("HideExplorerContextMenu", 1);
+                }
+            }
+        } catch (...) {}
+        delete Reg_HideExplorerContextMenu;
+    }

```

Comment:
See above (should not be needed).

Reply:
This registry key is read by the shell extension to enable hiding it.

Reply:
I see that with Qt you check shellExtension and not hide-something, I don't see why it is not possible to do the same here and why another key is needed. 

i.e.:

```
#ifdef MEDIAINFO_QT
            if (!RegGetBool(HKEY_CURRENT_USER, L"Software\\MediaArea.net\\MediaInfo", L"shellExtension"))
#else
            if (RegGetDword(HKEY_CURRENT_USER, L"Software\\MediaArea\\MediaInfo", L"HideExplorerContextMenu"))
#endif // MEDIAINFO_QT
```

Replaced by:
```
#ifdef MEDIAINFO_QT
            if (!RegGetBool(HKEY_CURRENT_USER, L"Software\\MediaArea.net\\MediaInfo", L"shellExtension"))
#else
            if (!RegGetDword(HKEY_CURRENT_USER, L"Software\\MediaArea\\MediaInfo", L"ShellExtension"))
#endif // MEDIAINFO_QT
```

Reply:
Software\\MediaArea\\MediaInfo\\ShellExtension does not exist currently. Therefore the VCL code needs to write the registry anyway. It's just the naming. I named it hide because after install, there will be no registry entries. So if hide key does not exist then the context menu entry is not hidden. So that context menu is available immediately after install.

Reply:
I was wrong, I thought currently it is available after install that's why. I'll change it. Forget about making it available before first launch, not necessary when using the installer interactively as it auto launches after install.

Reply:
The Qt one will have it enabled by default right after install though.

Reply:
> The Qt one will have it enabled by default right after install though.

I didn't see that but I am fine also with that, the registry key could be created also in the NSI.
But again, I prefer to split the PR in several smaller PRs, with the minimum in each PR.

Reply:
> the registry key could be created also in the NSI

That will only work for current user. But again, not necessary to have it by default in VCL version as mentioned above as it will auto launch after interactive install and auto open Pref window with shell extension checked by default. Qt one is different because it does not launch the Pref window on first launch and may not auto launch after install too.

Reply:
OK. I don't comment more the current PR, I let you change it / split it based on the discussion (I will quickly test and merge the PR which contains fixes not related to the Win11 integration so you don't have to maintain many branches at the same time).

Reply:
@JeromeMartinez I was not wrong after all. The current MediaInfo context menu is enabled after install without requiring MediaInfo to be run at all. This is after an uninstall and install.

Reply:
My mistake. But let's keep it simple here anyway, for a first shot, or maybe we can find something simpler e.g. registering the Registry key in the installer and launching the DLL which checks this value?